auth/cloudfrontAuthFunction.js

```diff
@@ -0,0 +1,40 @@
+// This is a CloudFront Viewer Request function that puts a simple authentication layer in from ouf our .ics files.
+// It expects a ?key be present in the URL for any file behind the CloudFront distribution, for example:
+// https://api.discal.app/guilds/890053994926460939/events.ics?key=XXXX
+
+// Any request that doesn't have a Guild ID in the expected part of the URL, or has an invalid key, will receive a 401 response.
+
+// This means nothing but these .ics files - in our /guilds/<id>/etc format - can ever be served from the domain the CF distro serves, but that's fine.
+
+// This code is deployed through a weird <textarea> in CloudFront > Functions.
+
+// See generateApiKeyForGuild.js for details on generated the key for any given Guild.
+
+var crypto = require('crypto');
+
+var response401 = {
+  statusCode: 401,
+  statusDescription: 'Unauthorized'
+};    
+
+function handler(event) {
+    if(!event.request.querystring.key){
+      return response401;
+    }
+    
+     // We can't setup this key with an env var or secret or anything, and i'm not going to commit it here.
+     // So before modifying this in CloudFront you'll have to enter the real key, which is DISCAL_AUTH_SECRET in .env.
+    var defcalSecret = "XXXXXXXX"
```

Comment:
That's a bummer. I wonder if we can set up a deploy script to do string replacement of env vars in this script prior to deploy, so we still get the benefits of env vars but without having to update files manually and avoid checking them in.

Reply:
Looks like Cloudflare Workers has support for env vars, and might not be too bad to move over there if we wanted to give that a try.

Reply:
Short of moving to Terraform, there's no deploy script to speak of. I just manually copy and pasted the code into the CloudFront config, see attached screen shot.

<img width="1137" alt="image" src="https://user-images.githubusercontent.com/71086/180437961-5f245fdd-02ad-4e28-86ae-31bb80eb597f.png">

Happy to move everything over to Cloudflare if you'd prefer.